### PR TITLE
Create br-submariner in each worker node for OVN to use

### PR DESCRIPTION
**Pull request by mangelajo**
Related-Issue: #1276

Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

**Comment by mangelajo** [comment]
Ok, there's one failure in OVN, but it's unrelated (hostnet->remote pod) which seems to happen sometimes in kind.

File: pkg/routeagent_driver/handlers/ovn/bridge.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+func (ovn *Handler) getMTU() (int, error) {
+	link, err := ovn.netlink.LinkByName(OVNK8sMgmntIntfName)
+
+	if err != nil {
+		if _, ok := err.(netlink.LinkNotFoundError); !ok {
+			return 0, errors.Wrapf(err, "error retrieving link by name %q", OVNK8sMgmntIntfName)
+		}
+	}
+
+	return link.Attrs().MTU, nil
```
Comment (sridhargaddam) [inline pkg/routeagent_driver/handlers/ovn/bridge.go]: If LinkNotFound, this would give an error.

**Review by aswinsuryan — APPROVED** [review]

File: pkg/routeagent_driver/handlers/ovn/bridge.go (hunk 1 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/ovn/bridge.go]: Oh right, thanks.

File: pkg/routeagent_driver/handlers/ovn/bridge.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+func (ovn *Handler) ensureSubmarinerNodeBridge() error {
+	MTU, err := ovn.getMTU()
```
Comment (tpantelis) [inline pkg/routeagent_driver/handlers/ovn/bridge.go]:
```suggestion
	mtu, err := ovn.getMTU()
```

File: pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+const ovsCommandTimeout = 15
+
+func vsctlCmd(parameters ...string) (output string, err error) {
+	allParameters := []string{fmt.Sprintf("--timeout=%d", ovsCommandTimeout)}
+	allParameters = append(allParameters, parameters...)
+
+	cmd := exec.Command("/usr/bin/ovs-vsctl", allParameters...)
+
+	klog.V(log.DEBUG).Infof("ovs-vsctl %v", allParameters)
```
Comment (tpantelis) [inline pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go]: TRACE?

File: pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	out, err := cmd.Output()
+
+	stdout := string(out)
+
+	if err != nil {
+		stdErrOut := stderr.String()
+		klog.Errorf("error running ovs-vsctl %+v, output:\n%s", err, stdErrOut)
```
Comment (tpantelis) [inline pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go]:
```suggestion
		klog.Errorf("Error running ovs-vsctl %+v, output:\n%s", err, stdErrOut)
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/routeagent_driver/handlers/ovn/bridge.go (hunk 1 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/ovn/bridge.go]: Done.

File: pkg/routeagent_driver/handlers/ovn/bridge.go (hunk 2 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/ovn/bridge.go]: Done.

File: pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go (hunk 3 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go]: Only happens at startup, but yeah let's move it to TRACE.

File: pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go (hunk 4 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/ovn/vsctl/vsctl.go]: Done

**Review by tpantelis — APPROVED** [review]

**Pull request merged**